internal/langserver/handlers/service.go

```diff
@@ -94,14 +94,17 @@ func (svc *service) Assigner() (jrpc2.Assigner, error) {
 
 	// The following is set via CLI flags, hence available in the server context
 	execOpts := &exec.ExecutorOpts{}
-	if path, ok := lsctx.TerraformExecPath(svc.srvCtx); ok {
-		execOpts.ExecPath = path
+	tfExecPath, ok := lsctx.TerraformExecPath(svc.srvCtx)
+	if ok {
+		execOpts.ExecPath = tfExecPath
 	} else {
-		tfExecPath, err := svc.tfDiscoFunc()
+		path, err := svc.tfDiscoFunc()
```

Comment:
I think that long term we could probably move this fallback discovery logic into `initialize` handler but we can address that in a separate PR.